crypto/rand/rand_lib.c

```diff
@@ -319,13 +319,15 @@ DEFINE_RUN_ONCE_STATIC(do_rand_init)
     if (rand_nonce_lock == NULL)
         goto err2;
 
-    if (!rand_cleaning_up && !rand_pool_init())
+    if (!rand_pool_init())
```

Comment:
One question, just to be sure (it's too late now for me to go through the details): Since you are effectively reverting bc420ebea2c5 here, did you check whether your pull request handles issue #7022 correctly?

Reply:
Yes.
I stepped thru test/errtest.c which does not initialize random.
set a breakpoint on rand_cleanup_int, and see this function
not initializing the rand_pool_init. and then rand_pool_cleanup
use uninitalized fd, and all that not happening with this patch,
because rand_inited is zero, and that is exactly what #7022 is about.